cupy/lib/polynomial.pyx

```diff
@@ -123,13 +126,11 @@ cdef class poly1d:
             return poly1d(self.coeffs * other)
         raise NotImplementedError
 
-    # TODO(Dahlia-Chehata): implement using polyadd
     def __add__(self, other):
-        raise NotImplementedError
-
-    # TODO(Dahlia-Chehata): implement using polyadd
-    def __radd__(self, other):
-        raise NotImplementedError
+        if isinstance(self, numpy.generic):
+            # for the case: numpy scalar + poly1d
+            return self + other._coeffs
```

Comment:
```suggestion
            return self + other.coeffs
```
To allow `bool` types, device synchronization is to be disabled

Reply:
Why device synchronization is required only for bool type inputs?

Reply:
It is not only for bool inputs. I meant to reply for why no_bool is needed in this particular test which BTW was irrelevant for other types. I have changed it. PTAL